### PR TITLE
shell: Include state.jsx in type checking

### PR DESCRIPTION
**Pull request by mvollmer**
This is a minimum effort to get test/static-code to find errors in ShellState. It doesn't do anything for users of ShellState.

### CODE REVIEW

File: pkg/shell/hosts.jsx [hunk 1]
```diff
@@ -130,8 +130,7 @@ export class CockpitHosts extends React.Component {
 
         const connection_string = machine.connection_string;
         const parts = split_connection_string(connection_string);
-        const addr = build_href({ host: parts.address });
-        state.jump(addr);
+        state.jump({ host: parts.address });
```
Comment (cockpituous) [inline pkg/shell/hosts.jsx:133]: This added line is not executed by any test.

File: pkg/shell/hosts.jsx [hunk 2]
```diff
@@ -147,8 +146,7 @@
 
         if (current_machine === machine) {
             // Removing machine underneath ourself - jump to localhost
-            const addr = build_href({ host: "localhost" });
-            state.jump(addr);
+            state.jump({ host: "localhost" });
```
Comment (cockpituous) [inline pkg/shell/hosts.jsx:149]: This added line is not executed by any test.

File: pkg/shell/hosts.jsx [hunk 3]
```diff
@@ -189,14 +187,14 @@
...
-                to={build_href({ host: m.address })}
+                href={encode_location({ host: m.address })}
                 active={m === current_machine}
                 key={m.key}
                 name={m.label}
                 header={(m.user ? m.user : this.state.current_user) + " @"}
                 status={m.state === "failed" ? { type: "error", title: _("Connection error") } : null}
                 className={m.state}
-                jump={() => this.onHostSwitch(m)}
+                onClick={() => this.onHostSwitch(m)}
```
Comment (cockpituous) [inline pkg/shell/hosts.jsx:197]: This added line is not executed by any test.

File: pkg/shell/hosts_dialog.jsx [hunk 4]
```diff
@@ -84,10 +84,9 @@ export async function edit_host(state, shell_state, machine) {
     const connection_string = await state.show_modal({ address: machine.address });
     if (connection_string) {
         const parts = split_connection_string(connection_string);
-        const addr = build_href({ host: parts.address });
         if (machine == current_machine && parts.address != machine.address) {
             shell_state.loader.connect(parts.address);
-            shell_state.jump(addr);
+            shell_state.jump({ host: parts.address });
```
Comment (cockpituous) [inline pkg/shell/hosts_dialog.jsx:89]: This added line is not executed by any test.

File: pkg/shell/nav.jsx [hunk 5]
```diff
@@ -145,10 +145,14 @@ export class CockpitNav extends React.Component {
...
                                 { g.action &&
-                                    <a className="pf-v5-c-nav__section-title nav-item" href={g.action.path} onClick={ ev => {
-                                        this.props.jump(g.action.path);
-                                        ev.preventDefault();
-                                    }}>{g.action.label}</a>
+                                    <a className="pf-v5-c-nav__section-title nav-item"
+                                        href={encode_location(g.action.target)}
+                                        onClick={ ev => {
+                                            this.props.jump(g.action.target);
+                                            ev.preventDefault();
```
Comment (cockpituous) [inline pkg/shell/nav.jsx:152]: These 5 added lines are not executed by any test.

File: pkg/shell/nav.jsx [hunk 6]
```diff
@@ -145,10 +145,14 @@
...
+                                        }}>
+                                        {g.action.label}
+                                    </a>
```
Comment (cockpituous) [inline pkg/shell/nav.jsx:155]: These 2 added lines are not executed by any test.

File: pkg/shell/nav.jsx [hunk 7]
```diff
@@ -356,10 +363,10 @@
     if (current_machine_manifest_items.items.apps && groups.length === 3)
         groups[0].action = {
             label: _("Edit"),
-            path: build_href({
+            target: {
```
Comment (cockpituous) [inline pkg/shell/nav.jsx:366]: This added line is not executed by any test.

File: pkg/shell/nav.jsx [hunk 8]
```diff
@@ -356,10 +363,10 @@
...
                 host: current_machine.address,
-                path: current_machine_manifest_items.items.apps.path
-            })
+                path: current_machine_manifest_items.items.apps.path,
```
Comment (cockpituous) [inline pkg/shell/nav.jsx:368]: This added line is not executed by any test.

File: pkg/shell/state.tsx [hunk 9]
```diff
@@ -57,13 +61,13 @@ export function ShellState() {
 
     machines.addEventListener("ready", on_ready);
 
-    machines.addEventListener("removed", (ev, machine) => {
+    machines.addEventListener("removed", (_, machine) => {
```
Comment (cockpituous) [inline pkg/shell/state.tsx:64]: This added line is not executed by any test.

File: pkg/shell/util.tsx [hunk 10]
```diff
@@ -19,57 +19,50 @@
...
+export interface Location {
+    host: string;
+    path: string;
+    hash: string;
+}
+
+export function encode_location(location: Location): string {
+    const shell_embedded = window.location.pathname.indexOf(".html") !== -1;
+    if (shell_embedded)
+        return window.location.toString();
```
Comment (cockpituous) [inline pkg/shell/util.tsx:31]: This added line is not executed by any test.

**Comment by mvollmer** [comment]
Nope.

**Pull request closed without merging**